src/file_pipe_log/pipe_builder.rs

```diff
@@ -300,7 +306,7 @@ impl<F: FileSystem> DualPipesBuilder<F> {
                     let is_last_file = index == chunk_count - 1 && i == file_count - 1;
                     match build_file_reader(file_system.as_ref(), f.handle.clone(), None) {
                         Err(e) => {
-                            let is_local_tail = f.handle.file_size()? <= LogFileFormat::len();
+                            let is_local_tail = f.handle.file_size()? <= default_format_len;
```

Comment:
This is problematic. Opening with V2 and reading corrupted V1 file would not raise an error.

I think you can modify `build_file_reader` to return a `Result<Option<Reader>>`, a return value of `Ok(None)` means we hit EOF.

Reply:
I prefer to use a special `Error::InvalidArguement` to mark this error, to avoid making batch code revisions on the relevant calling paths.

Reply:
Add a test.

Reply:
Yes, it already been supplied in the previous commit, where test with `fail_points` on corrupted `DataLayout` has passed. It can be reviewed in `failpoints/test_engine.rs::test_tail_corruption`.

I'm still confused that why the code path on `Err(Error::InvalidArguement(...))` is not covered, but the code path on examing `Error::InvalidArguement` in `reader.rs` is covered. Also, the print on it also could prove that it already makes sense: 

``` Rust
// in file_pipe_log/format.rs
} else {
        // Here, we mark this special err, that is, corrupted `payload`,
        // with InvalidArgument.
        println!("invalid dataload in the header, len: {}, expected len: {}",
                buf_len - Self::header_len(), Self::payload_len(version));
        Err(Error::InvalidArgument(format!(
            "invalid dataload in the header, len: {}, expected len: {}",
            buf_len - Self::header_len(), Self::payload_len(version)
        )))
    }
```
And the `failpoints/test_engine.rs::test_tail_corruption` get the following prints with expectation:
> ``` invalid dataload in the header, len: 7, expected len: 8```

Reply:
Covering a specific line of code is different than covering a code path (the LOC you added could be encountered in many different situations, and we want to cover them all). What I want is to add a test that checks "Opening with V2 and reading corrupted V1 file would raise an error".

Reply:
The relevant test cases have been supplemented. And the previous bug, checking log file with corrupted V1 header with `file_size() <= header_len() + payload_len()`, have been corrected by reverting the changes.